src/main/java/org/folio/services/financedata/FinanceDataService.java

```diff
@@ -1,39 +1,191 @@
+                                                    RequestContext requestContext) {
+    var transactionsFuture = fyFinanceDataCollection.getFyFinanceData().stream()
+      .map(financeData -> createAllocationTransaction(financeData, requestContext))
+      .toList();
+
+    return GenericCompositeFuture.join(transactionsFuture)
+      .compose(compositeFuture -> {
+        List<Transaction> transactions = compositeFuture.list();
+        return createBatchTransaction(transactions, requestContext);
+      });
+  }
+
+  private Future<Transaction> createAllocationTransaction(FyFinanceData financeData, RequestContext requestContext) {
+    var allocation = calculateAllocation(financeData);
+    var transaction = new Transaction()
+      .withTransactionType(Transaction.TransactionType.ALLOCATION)
+      .withId(UUID.randomUUID().toString())
+      .withAmount(allocation)
+      .withFiscalYearId(financeData.getFiscalYearId())
+      .withToFundId(financeData.getFundId())
+      .withSource(Transaction.Source.USER);
+    log.info("Creating allocation transaction for fund '{}' and budget '{}' with allocation '{}'",
+      financeData.getFundId(), financeData.getBudgetId(), allocation);
+
+    return fiscalYearService.getFiscalYearById(financeData.getFiscalYearId(), requestContext)
+      .map(fiscalYear -> transaction.withCurrency(fiscalYear.getCurrency()));
+  }
+
+  private Double calculateAllocation(FyFinanceData financeData) {
+    var initialAllocation = BigDecimal.valueOf(financeData.getBudgetInitialAllocation());
+    var allocationChange = BigDecimal.valueOf(financeData.getBudgetAllocationChange());
+    return initialAllocation.add(allocationChange).doubleValue();
+  }
+
+  public Future<Void> createBatchTransaction(List<Transaction> transactions, RequestContext requestContext) {
+    Batch batch = new Batch().withTransactionsToCreate(transactions);
+    return transactionApiService.processBatch(batch, requestContext);
+  }
+
+  private Future<Void> updateFinanceData(FyFinanceDataCollection financeDataCollection,
+                                         RequestContext requestContext) {
+    log.debug("updateFinanceData:: Trying to update finance data collection with size: {}", financeDataCollection.getTotalRecords());
+    return restClient.put(resourcesPath(FINANCE_DATA_STORAGE), financeDataCollection, requestContext);
+  }
+
+  private void processLogs(FyFinanceDataCollection financeDataCollection,
+                           RequestContext requestContext, FundUpdateLog.Status status) {
+    var jobDetails = new JobDetails().withAdditionalProperty("fyFinanceData", financeDataCollection.getFyFinanceData());
+    var fundUpdateLog = new FundUpdateLog().withId(UUID.randomUUID().toString())
+      .withJobName("Update finance data")
+      .withStatus(status)
+      .withRecordsCount(financeDataCollection.getTotalRecords())
+      .withJobDetails(jobDetails)
+      .withJobNumber(1);
```

Comment:
we will have separate to handle job numbers and can use an appoach as in mod-orders-storage with separate table. See mod-orders-storage OrdersDao.getNextPoNumber() for reference after confirming with PO that this functionality is needed